mantidproject/mantid
# Allow MDHistoWorkspace to be produced from NormPoly transformation

Fixes #13602 

ConvertToReflectometryQ should now have no restrictions on whether or not you can output an MD Workspace when performing a NormPoly Transform.

**To Test**:

<b> Producing MDWorkspace </b>
- Load [data](https://slack-files.com/T02J4MMEW-F0AA5NGSY-405b51aad0) in Mantid
- Run this code through script interpreter:

``` python
out_ws, dump_vertexes = ConvertToReflectometryQ(InputWorkspace='data_th_lam',OutputWorkspace='QxQy_polyMD', OutputDimensions='Q (lab frame)',
Extents='-0.0005,0.0005,-0,0.2', OutputAsMDWorkspace=True,Method='NormalisedPolygon',  IncidentTheta=0.44, OverrideIncidentTheta=True, NumberBinsQx=100, NumberBinsQz=100,DumpVertexes=True, OutputVertexes='dump_vertexes')
```
- See that an MDHistoWorkspace called `QxQy_polyMD` is produced

<b> Produce 2D Workspace </b>
- Load [data](https://slack-files.com/T02J4MMEW-F0AA5NGSY-405b51aad0) in Mantid
- Run this code through script interpreter:

``` python
out_ws, dump_vertexes = ConvertToReflectometryQ(InputWorkspace='data_th_lam',OutputWorkspace='QxQy_poly2D', OutputDimensions='Q (lab frame)',
Extents='-0.0005,0.0005,-0,0.2', OutputAsMDWorkspace=False,Method='NormalisedPolygon',  IncidentTheta=0.44, OverrideIncidentTheta=True, NumberBinsQx=100, NumberBinsQz=100,DumpVertexes=True, OutputVertexes='dump_vertexes')
```
- See that a 2DWorkspace `QxQy_poly2D` is produced

<b> Comparing Plots </b>
- Open the MDHistoWorkspace in SliceViewer
- Open the 2DWorkspace in SliceViewer
- Compare the plots to make sure they are identical.
